app/javascript/components/import-image-form.schema.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { componentTypes, validatorTypes } from '@@ddf';
+
+
+const createSchema = (providers, provider, setProvider, images, image, setImage, storages, storage, setStorage, buckets, bucket, setBucket, diskTypes, setDiskType, keepOva, setKeepOva) => ({
+  fields: [
+    {
+      component: componentTypes.SELECT,
+      name:"src_provider_id",
+      id:"src_provider_id",
+      initializeOnMount: true,
+      isRequired: true,
+      label: "Choose source provider",
+      loadOptions: () => providers,
+      onChange: (value) => {setProvider(value); setImage('-1')},
```

Comment:
I really couldn't understand why we need to set values for provider and below fields, these will be available in values.src_provider_id,values.src_image_id....., You can refer https://github.com/ManageIQ/manageiq-ui-classic/tree/master/app/javascript/components/subnet-form or any other form for reference.

Reply:
I've refactored this part doing it in a similar fashion as in the manageiq-ui-classic module. I still had to set the state in the onChange handler's of two fields though. The reason being is that the next fields depend on the choice done in previous ones. They're dynamic and hence the chosen values are handled not only in the "submit" function where the "values" field is available.

Reply:
done as well, as discussed during the telco